src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final double MAX_RANDOM_VALUE = 10.;
+    private static final int AMOUNT_OF_FIGURES = 4;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        int index = random.nextInt(AMOUNT_OF_FIGURES);
+        switch (index) {
+            case 0:
+                return new Square(color, randomSide());
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create separate variables for the sides and pass them on for better code readability. This applies to all cases in the switch statement where you're directly passing the result of `randomSide()` to the constructor.